drivers/sensor/ina23x/ina23x.c

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright 2021 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_ina23x
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include "ina23x.h"
+
+/**
+ * @brife Macro for setting the current_lsb to 1mA
+ */
+#define CURRENT_LSB_1MA 1
+
+/**
+ * @brief Macro for creating the INA23X calibration value
+ *        CALIB = (5120 / (current_lsb * rshunt))
+ *        NOTE: The 5120 value is a fixed value internal to the
+ *              INA23X that is used to ensure scaling is properly
+ *              maintained.
+ *
+ * @param current_lsb  Value of the Current register LSB in milliamps
+ * @param rshunt       Shunt resistor value in milliohms
+ */
+ #define INA23X_CALIB(current_lsb, rshunt) (5120 / ((current_lsb) * (rshunt)))
+
+/**
+ * @brief Macro to convert raw Bus voltage to millivolts
+ *
+ * reg  value read from bus voltage register
+ */
+#define INA23X_BUS_MV(reg) ((reg) * 125 / 100)
+
+/**
+ * @brief Macro to convert raw Shunt voltage to microvolts
+ *
+ * reg  value read from shunt voltage register
+ */
+#define INA2XX_SHUNT_UV(reg) ((reg) * 25 / 10)
+
+/**
+ * @brief Macro to convert raw power value to milliwatts
+ *
+ * reg  value read from power register
+ */
+#define INA23X_POW_MW(reg) ((reg) * 25)
+
+static int ina23x_reg_read(const struct device *dev, uint8_t reg, int16_t *val)
+{
+	const struct ina23x_config *const config = dev->config;
+	struct ina23x_data *ina23x = dev->data;
+	uint8_t data[2];
+
+	if (i2c_burst_read(ina23x->i2c, config->bus_addr, reg, data, 2) < 0) {
+		LOG_ERR("Unagle to read register");
+		return -EIO;
+	}
+
+	*val = (data[0] << 8) | data[1];
+
+	return 0;
+}
+
+static int ina23x_reg_write(const struct device *dev, uint8_t reg, uint16_t val)
+{
+	const struct ina23x_config *const config = dev->config;
+	struct ina23x_data *ina23x = dev->data;
+	uint8_t tx_buf[3] = { reg, (val >> 8), (val & 0xff) };
+
+	if (i2c_write(ina23x->i2c, tx_buf, sizeof(tx_buf), config->bus_addr) < 0) {
+		LOG_ERR("Failed to write register");
```

Comment:
this log doesn't provide much information here (i2c driver logging can likely provide the same info)